src/renderers/shaders/ShaderChunk/shadowmap_pars_fragment.glsl

```diff
@@ -1,12 +1,17 @@
 #ifdef USE_SHADOWMAP
 
+	uniform int isShadowCube[ MAX_SHADOWS ];
+	uniform samplerCube shadowCube[ MAX_SHADOWS ];
 	uniform sampler2D shadowMap[ MAX_SHADOWS ];
 	uniform vec2 shadowMapSize[ MAX_SHADOWS ];
 
 	uniform float shadowDarkness[ MAX_SHADOWS ];
 	uniform float shadowBias[ MAX_SHADOWS ];
 
+	uniform vec3 shadowLightPosition[ MAX_SHADOWS ];
+
 	varying vec4 vShadowCoord[ MAX_SHADOWS ];
```

Comment:
> Ah, so are you thinking of doing away with 'shadowMatrix'?

I think we should avoid requiring higher limits. Especially in case the new feature is not being used. There is always the option of using a define to ensure the latter. It may be a good idea in any case to get rid of the cube sampler (*1) when it's not needed, however, when that's the only purpose I'd say we have a smooth integration.

The story of GL ES can be summarized as follows: `MAX_VARYING_VECTORS` can be as low as 8 and `MAX_FRAGMENT_UNIFORM_VECTORS` can be as low as 16. Only *statically referenced* variables count, meaning if you don't touch the variable they won't be allocated (the cleverness to detect dead code may vary across different drivers). In theory, there should also be some automatic packing, which we can't rely on in practice. 

> Without it, we get 4 light positions for "free", right?

No. I was wrong here. Using a matrix in the vertex shader is not much of a problem, however,

>Additionally, I would be re-purposing 'vShadowCoord' to hold the world-space position of the light. I admit it's not the most readable code, but I'd make sure to comment appropriately.

adding extra varyings is. => Using `vShadowCoord` seems a very good idea.

Comments are nice, and even nicer is to "say it in code" when possible (comments easily get outdated by edits, code OTOH has to be dealt with), e.g.:

```javascript
bool isPointLight = shadowDarkness[ i ] < 0;
// ...
```